src/Components/Blazor/Build/src/targets/Blazor.MonoRuntime.targets

```diff
@@ -318,12 +306,24 @@
     Inputs="@(BlazorOutputWithTargetPath)"
     Outputs="$(BlazorBootJsonIntermediateOutputPath)">
     <ItemGroup>
-      <_BlazorRuntimeFile Include="@(BlazorOutputWithTargetPath->WithMetadataValue('BlazorRuntimeFile', 'true'))" />
+      <_BlazorBootResource Include="@(BlazorOutputWithTargetPath->WithMetadataValue('BlazorRuntimeFile', 'true'))" />
+      <_BlazorBootResource BootResourceType="assembly" Condition="'%(Extension)' == '.dll'" />
+      <_BlazorBootResource BootResourceType="pdb" Condition="'%(Extension)' == '.pdb'" />
+      <_BlazorBootResource BootResourceType="wasm" Condition="'%(Extension)' == '.wasm'" />
```

Comment:
Should we be concerned about files that don't match one of these extension types? Are the non-matching files all of the *other* static assets that don't belong in boot.json?

Reply:
The ones without the BlazorRuntimeFile are the other static assets that don’t belong in the boot json. 

Additionally we’re excluding other extensions from boot json by virtue of not assigning any BootResourceType. This eliminates the AssemblyName.xml files that would otherwise sometimes appear.

Reply:
Why do we think it is important to differentiate by dll, pdbs and wasm? I think having the `Condition` piece should be good enough.

Reply:
The logic about how to distinguish the file types has to exist somewhere. I agree it could be done on the .NET side too but that seems neither better nor worse. If you can anticipate some problem with doing it on the MSBuild side I'm fine with changing it, but otherwise I suspect it's a neutral choice and we're just as well leaving it as-is. It's an internal implementation detail we can change later if we want.